src/Application/Models/Bank.js

```diff
@@ -28,15 +28,31 @@ class Bank {
     );
   }
 
-  withdraw(cost, newIncome) {
+  withdraw(amountToWithdraw) {
+    if (amountToWithdraw > this._cash) {
+      console.warn(new Error(
+        'amountToWithdraw high than current cash.',
+        'amountToWithdraw:', amountToWithdraw,
+        'this: ', this
+      ));
+    }
     return new Bank(
-        this._cash - cost,
-        newIncome,
+        this._cash - amountToWithdraw,
+        this._income,
         this._total,
         this._lastRentTime
     );
   }
 
+  updateIncome(newIncome) {
```

Comment:
This was overdue, but there might be some issues based off of the way that I organized the COLLECT_INCOME action.

Redux has a simple way to chain reducers and now that we have hit this point its time for me to dive into that, but the crux of the issue is that I didn't want to recalculate the income from buildings everytime I COLLECT_INCOME, so that value is stored and specifically updated on a BUY. Let's talk offline.